exchange/exchange.go

```diff
@@ -0,0 +1,66 @@
+package exchange
+
+import (
+	"context"
+	"sync"
+	"time"
+)
+
+//go:generate go run github.com/vektra/mockery/v2@v2.20.2 --name Exchanger
+type Exchanger interface {
+	PublishMessage(queueName string, message string) bool
+	ReceiveMessage(ctx context.Context, queueName string, waitTime time.Duration) (string, error)
+}
+
+type Exchange struct {
+	lock   sync.Mutex
+	queues map[string]*queue[string]
+}
+
+// NewExchange Returns new exchange
+func NewExchange() *Exchange {
+	return &Exchange{
+		queues: make(map[string]*queue[string]),
+	}
+}
+
+// getQueue Thread-safe get queue from map
+func (e *Exchange) getQueue(queueName string) *queue[string] {
+	e.lock.Lock()
+	defer e.lock.Unlock()
+	q, ok := e.queues[queueName]
+
+	if !ok {
```

Comment:
Так визуально лучше разделено где мок берем, где мапу читаем
```suggestion
	e.lock.Lock()
	defer e.lock.Unlock()
	
	q, ok := e.queues[queueName]
	if !ok {
```